src/TriangleNodesProvisioner.cpp

```diff
@@ -560,6 +563,40 @@ namespace blitzdg {
             mB(i) = tmpMapB(i);
             vmB(i) = vmM(mB(i));
         }
+
+        buildBCHash();
+    }
+
+    void TriangleNodesProvisioner::buildBCHash() {
+        firstIndex ii;
+        secondIndex jj;
+        index_vector_type bcVec = Mesh2D.get_BCType();
+
+        // allocate correct storage for boundary node indices of each type.
+        index_hashmap& bcMap = *BCmap;
+
+        // create boundary face nodes global numbering.
+        index_matrix_type boundaryNodesMat(NumFacePoints, NumFaces*NumElements);
+
+        index_vector_type ones(NumFacePoints);
+        ones = 0*ii + 1;
+        boundaryNodesMat = ones(ii)*bcVec(jj);
+
+        index_type count=0;
+        for (index_type f=0; f < NumFaces*NumElements; ++f) {
+            for (index_type n=0; n < NumFacePoints; ++n) {
+                index_type bct = boundaryNodesMat(n, f);
```

Comment:
Since `boundaryNodesMat` is row-oriented you should have the n-loop as the outer loop and the f-loop as the inner loop. Alternatively, a better approach would be to use an iterator to iterate over the elements of `boundaryNodesMat` in a row-contiguous manner.

Reply:
Something like the following should work:
```
for (auto itr = boundaryNodesMat.begin(); itr != boundaryNodesMat.end(); ++itr)
```